packages/mobile/src/exchange/WithdrawCeloScreen.tsx

```diff
@@ -0,0 +1,183 @@
+// and an amount.
+
+import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button.v2'
+import KeyboardAwareScrollView from '@celo/react-components/components/KeyboardAwareScrollView'
+import KeyboardSpacer from '@celo/react-components/components/KeyboardSpacer'
+import TextInputWithButtons from '@celo/react-components/components/TextInputWithButtons'
+import Touchable from '@celo/react-components/components/Touchable'
+import QRCodeBorderlessIcon from '@celo/react-components/icons/QRCodeBorderless'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import variables from '@celo/react-components/styles/variables'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import React, { useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, TouchableOpacity } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { CeloExchangeEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import ClipboardAwarePasteIcon from 'src/components/ClipboardAwarePasteIcon'
+import { CURRENCY_ENUM } from 'src/geth/consts'
+import i18n, { Namespaces } from 'src/i18n'
+import { HeaderTitleWithBalance, headerWithBackButton } from 'src/navigator/Headers.v2'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import useSelector from 'src/redux/useSelector'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+
+type Props = StackScreenProps<StackParamList, Screens.WithdrawCeloScreen>
+
+function WithdrawCeloScreen({ navigation }: Props) {
+  const [accountAddress, setAccountAddress] = useState('')
+  const [celoToTransfer, setCeloToTransfer] = useState('')
+
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+  const goldBalanceNumber = new BigNumber(goldBalance || 0)
+  const { t } = useTranslation(Namespaces.exchangeFlow9)
+
+  const celoToTransferNumber = new BigNumber(celoToTransfer)
+  const readyToReview =
+    accountAddress.startsWith('0x') &&
+    accountAddress.length === 42 &&
+    celoToTransferNumber.isGreaterThan(0) &&
+    celoToTransferNumber.isLessThanOrEqualTo(goldBalanceNumber)
+
+  const onConfirm = useCallback(() => {
+    const celoAmount = new BigNumber(celoToTransfer)
+    ValoraAnalytics.track(CeloExchangeEvents.celo_withdraw_review, {
+      amount: celoAmount.toString(),
+    })
+    navigation.navigate(Screens.WithdrawCeloReviewScreen, {
+      amount: celoAmount,
+      recipientAddress: accountAddress,
+    })
+  }, [accountAddress, celoToTransfer])
+
+  const setMaxAmount = useCallback(() => {
+    if (goldBalance) {
+      setCeloToTransfer(goldBalance)
+    }
```

Comment:
could some of this logic be moved into a custom hook? 

call be old fashion but I think that its easier to follow function when they are fewer lines

Reply:
These are very small functions, I doubt, that moving the logic from them somewhere else would benefit readability.

Reply:
I moved the inputs and input logic to custom components, I think it looks much better now :)